src/main/scala/ReproducibleBuildsPlugin.scala

```diff
@@ -101,12 +95,62 @@ object ReproducibleBuildsPlugin extends AutoPlugin {
       reproducibleBuildsPackageName.value,
       ext,
       ext,
-      "compile",
+      Compile.name,
       extraModuleAttributes,
       null
     )
   }
 
+  def locationFromResolver(r: DependencyResolver, ext:String) = Def.task {
+    import SbtLibraryManagementFunctions._
+    val moduleInfo = (moduleSettings.value match {
+      case ic: InlineConfiguration =>
+        val icWithCross = substituteCross(ic)
+        if (sbtPlugin.value) appendSbtCrossVersion(icWithCross)
+        else icWithCross
+    }).moduleInfo
+
+    val artifactOrigin = r.locate(
+      toIvyArtifact(
+        newConfiguredModuleID(moduleID.value, moduleInfo, Vector(Compile)),
+        artifact.value.withExtension(ext),
+        Vector(Compile)
+      )
+    )
+    artifactOrigin.getLocation
+  }
+
+  def artifactUrl(resolver: Resolver, ext: String) = Def.taskDyn {
+    import SbtLibraryManagementFunctions._
+
+    resolver match {
+      case repository: MavenRepository =>
+        val pattern = resolvePattern(
+          repository.root,
+          "[organisation]/[module](_[scalaVersion])/[revision]/[artifact](_[scalaVersion])-[revision](-[classifier]).[ext]"
+        )
+        substitutePattern(pattern, ext)
+      case repository: PatternsBasedRepository =>
+        val pattern = repository.patterns.artifactPatterns.headOption.orElse(
+          repository.patterns.ivyPatterns.headOption
+        ).getOrElse(
+          throw new IllegalArgumentException("Expected at least a single artifact pattern")
+        )
+        substitutePattern(pattern, ext)
+      case _: ChainedResolver =>
+        throw new IllegalArgumentException("Not yet implemented")
```

Comment:
While it is possible to make this `ChainResolver` work it would require copying a huge amount of private code in sbt-library-management.